File: back/schemas/reservation.py
from pydantic import BaseModel
from datetime import datetime


class ReservationBase(BaseModel):
    reader_id: int
    book_id: int
    library_id: int
    status: str = "active"  # active, completed, cancelled


class ReservationCreate(BaseModel):
    book_id: int
    library_id: int


class ReservationUpdate(BaseModel):
    status: str | None = None


class ReservationResponse(ReservationBase):
    id: int
    reservation_date: datetime

    model_config = {"from_attributes": True}


class ReservationWithDetails(ReservationResponse):
    book_title: str | None = None
    reader_name: str | None = None
    library_name: str | None = None

File: LibraSmartWPF/Models/Reservation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraSmartWPF.Models;

/// <summary>
/// Бронирования книг
/// </summary>
[Table("reservations")]
public class Reservation
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("reader_id")]
    public int ReaderId { get; set; }

    [Required]
    [Column("book_id")]
    public int BookId { get; set; }

    [Required]
    [Column("library_id")]
    public int LibraryId { get; set; }

    [Column("reservation_date")]
    public DateTime ReservationDate { get; set; } = DateTime.UtcNow;

    [Required]
    [MaxLength(20)]
    [Column("status")]
    public string Status { get; set; } = "active"; // active, completed, cancelled

    // Связи
    [ForeignKey(nameof(ReaderId))]
    public virtual Reader Reader { get; set; } = null!;

    [ForeignKey(nameof(BookId))]
    public virtual Book Book { get; set; } = null!;

    [ForeignKey(nameof(LibraryId))]
    public virtual Library Library { get; set; } = null!;
}
